### PR TITLE
ransack and kaminari for searching, sorting, filtering and pagination

### PR DESCRIPTION
**Pull request by agarwalrashika00**

### CODE REVIEW

File: app/views/admin/users/index.html.erb [hunk 1]
```diff
@@ -1,32 +1,73 @@
-<%= link_to 'Create new user', new_admin_user_path, class: 'btn btn-dark mt-2 mr-2 float-right' %>
-<h2 class="text-center mt-2">All Users</h2>
-<div class="container-fluid">
-  <div class="row justify-content-center">
+<%= link_to '&plus;'.html_safe + ' Create new', new_admin_user_path, class: 'btn btn-dark mt-2 mr-2 float-right border' %>
```
Comment (Ankita9818) [inline app/views/admin/users/index.html.erb:1]: Lets use icons instead of ascii codes

File: app/views/kaminari/_first_page.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<li class="page-item">
```
Comment (Ankita9818) [inline app/views/kaminari/_first_page.html.erb:1]: Why these?? Do we need to create them separately??

**Review by Ankita9818 — CHANGES_REQUESTED** [review]

File: config/initializers/kaminari_config.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+Kaminari.configure do |config|
+  config.default_per_page = 2
+  # config.max_per_page = nil
+  # config.window = 4
+  # config.outer_window = 0
+  # config.left = 0
+  # config.right = 0
+  config.page_method_name = :page
```
[the remark this answers is not in the document]
Comment (agarwalrashika00) [reply config/initializers/kaminari_config.rb:10]: Those are the dafault name and even if we donot set them we can still use those method names.

File: config/initializers/ransack.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
```
[the remark this answers is not in the document]
Comment (agarwalrashika00) [reply config/initializers/ransack.rb:1]: no not necessary, it basically freezes the string in the file and rubocop asks us to add the magic comment.

File: app/views/kaminari/_first_page.html.erb (hunk 2 above)
Comment (agarwalrashika00) [reply app/views/kaminari/_first_page.html.erb:1]: image
It looks like this if i donot add view files.

image
and it gets bootstrap classes if i add the view files

**Review by Ankita9818 — APPROVED** [review]

**Pull request merged**